src/client/views/book/bookEditorInterface.ml

```diff
@@ -179,6 +179,88 @@ let create ?on_save page =
+  let content = Html.createDiv (Page.document page) in
+  let title = Text.Heading.h2_static
+      ~text:(Lwt.return "Edit book") page in
+  let form = Html.createForm (Page.document page) in
+  let input_title = Inputs.Text.create
+      ~placeholder:(BookEditor.title editor)
+      ~on_change:(fun title -> BookEditor.set_title editor title)
+      page
+  in
+  let input_date = Inputs.Text.create
+      ~placeholder:(BookEditor.date editor)
+      ~on_change:(fun date -> BookEditor.set_date editor date)
+      page
+  in
+
+  let submit = Html.createDiv (Page.document page) in
+  Style.set ~display:"flex" submit;
+  submit##.classList##add (js "justify-content-space-between");
+
+  let sets_area = Html.createDiv (Page.document page) in
+  let sets_search =
+    let main_section =
+      SearchBar.Section.create
+        ~search:(fun input ->
+            let%rlwt formula = Lwt.return (SetFilter.raw input) in
+            let%lwt results =
+              Set.search ~threshold:0.4
+                ~pagination:Pagination.{start = 0; end_ = 10} formula
+            in
+            Lwt.return_ok results)
+        ~make_result:(fun score -> make_set_search_result editor page score)
+        page
+    in
+    SearchBar.create
+      ~placeholder:"Add set (Magic Search)"
+      ~sections:[main_section]
+      page
+  in
+
+  let t = {page; editor; content; input_title; input_date; sets_area; sets_search} in
+
+  let save =
+    Inputs.Button.create ~kind:Inputs.Button.Kind.Success ~icon:"save" ~text:"Save"
+      ~on_click:(fun () ->
+          let b1, b2 =
+            Inputs.Text.check input_title (fun str -> str <> ""),
+            Inputs.Text.check input_date
+              (fun str -> try PartialDate.from_string str |> ignore; true with _ -> str = "")
+
+          in
+          if b1 && b2 then (
+            Lwt.on_success (BookEditor.update_submit editor slug) (fun _ ->
+                begin match on_save with
+                  | None ->
+                    let href = Router.(path (Book slug)) in
+                    Html.window##.location##.href := js href
+                  | Some cb -> cb slug
+                end)))
+      page
+  in
+
+  Dom.appendChild submit (Inputs.Button.root save);
+
+  Dom.appendChild form (Inputs.Text.root input_title);
+  Dom.appendChild form (Html.createBr (Page.document page));
+  Dom.appendChild form (Inputs.Text.root input_date);
+  Dom.appendChild form (Html.createBr (Page.document page));
+  Dom.appendChild form sets_area;
+  Dom.appendChild form (Html.createBr (Page.document page));
+  Dom.appendChild form (SearchBar.root sets_search);
+  Dom.appendChild form (Html.createBr (Page.document page));
+  Dom.appendChild form submit;
+
+  Dom.appendChild content (Text.Heading.root title);
+  Dom.appendChild content (Html.createHr (Page.document page));
+  Dom.appendChild content (Html.createBr (Page.document page));
+  Dom.appendChild content form;
```

Comment:
This really calls for an `appendChildren` helper :p But the goal was to eventually deprecate that kind of things and replace them only by `Dancelor_client_html`.

Reply:
cf https://github.com/paris-branch/dancelor/issues/210